library/core/src/cmp.rs

```diff
@@ -506,9 +506,19 @@ impl<T: Ord> Ord for Reverse<T> {
 /// ## Derivable
 ///
 /// This trait can be used with `#[derive]`. When `derive`d on structs, it will produce a
-/// lexicographic ordering based on the top-to-bottom declaration order of the struct's members.
+/// [lexicographic](https://en.wikipedia.org/wiki/Lexicographic_order) ordering based on the top-to-bottom declaration order of the struct's members.
 /// When `derive`d on enums, variants are ordered by their top-to-bottom discriminant order.
 ///
+/// ## Lexicographical comparison
+///
+/// Lexicographical comparison is a operation with the following properties:
+///  - Two ranges are compared element by element.
```

Comment:
it is strange to use "range" here. in Rust "range" means the `a..b` object (std::ops::Range), or a selection within an ordered collection involving two end points (BTreeSet::range, VecDeque::range). 

perhaps you mean "arrays" or "tuples".

Reply:
I think this does not only refer to "arrays" or "tuples". I don't know if the term "iterable" is correct. Do you have any suggestions?

Reply:
Maybe ranges is OK, because I am not disturbed by std::ops::Range, I can understand its meaning.

Reply:
@Rustin-Liu as this passage is copied directly from [cppreference.com](https://en.cppreference.com/w/cpp/algorithm/lexicographical_compare), the term "range" has to be taken in the context on C++. in C++ (and D) a [range](https://en.cppreference.com/w/cpp/ranges) is defined to be a pair of iterators. This also makes sense for C++'s `std::lexicographical_compare` since it takes 2 pairs of iterators as input.

however, this documentation here is talking about the `Ord` trait in general, so using "range" here even using C++ semantics is incorrect.

if we use the [Wikipedia page](https://en.wikipedia.org/wiki/Lexicographic_order) you've linked we can use "sequence".

Reply:
Got it. Thanks!